### PR TITLE
Remove unicode-named test files

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Comes from https://github.com/conda-forge/python-libarchive-c-feedstock/issues/35

### CODE REVIEW

**Comment by pkgw** [comment]
It seems that the Windows approach didn't work here because something about the Unicode characters was crashing conda-build as it tried to write the build script file. I've pushed a small change that uses a Python helper script to avoid that.

But what I do find is that the Linux builds now fail, when they didn't before, with a test that relied on the nefarious files. I don't know why the Linux builds worked before, but it looks like we just need to make sure skip the relevant test(s) on all platforms.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,8 +12,8 @@ build:
   script: 
     - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
     # https://github.com/conda-forge/python-libarchive-c-feedstock/issues/35
-    - rm -f {{ SRC_DIR }}/tests/data/프로그램.*  # [unix]
-    - del /F /Q {{ SRC_DIR }}\\tests\\data\\프로그램.*  # [win]
+    - {{ PYTHON }} {{ RECIPE_DIR }}/delete_unicode_files.py  # [unix]
+    - {{ PYTHON }} {{ RECIPE_DIR }}\\delete_unicode_files.py  # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: FWIW could consolidate to something that works on either OS

```suggestion
    - {{ PYTHON }} {{ os.sep.join([RECIPE_DIR, "delete_unicode_files.py"]) }}
```

<hr>

Another OS agnostic option might be

```suggestion
    - cd {{ RECIPE_DIR }}
    - {{ PYTHON }} delete_unicode_files.py
```

Comment (pkgw) [reply recipe/meta.yaml]: The duplication bothered me too. I don't have a strong feeling about which approach is cleaner — either would be fine with me.

Comment (jakirkham) [reply recipe/meta.yaml]: Added another option above

Also could just [add this path to `source`]( https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#source-from-a-local-path ), which would work too

No strong feelings on any of these

Comment (jaimergp) [reply recipe/meta.yaml]: Oh, thanks! Added those suggestions. Sorry I messed the co-author metadata, John.

Comment (jakirkham) [reply recipe/meta.yaml]: Thanks Jaime! 🙏 

No worries. HTH 🙂

**Comment by jaimergp** [comment]
Yay, that worked! 🚀 Thank you all!

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

**Comment by pkgw** [comment]
Um, are we ready to go ahead and merge this?

**Comment by jaimergp** [comment]
I think so @pkgw!

**Pull request merged**